lib/vector-core/src/time.rs

```diff
@@ -0,0 +1,15 @@
+//! Time utilities for vector-core
+
+use std::task::{Context, Poll};
+
+/// A trait for representing timers
+///
+/// Embedding time as a type into other types eases property testing and
+/// verification. As such, this simple time type encodes the notion of elapsed
+/// timing with reset.
+pub trait Timer {
+    // For an example of how property testing can use this type see the
+    // `stream::Batcher` property tests.
+    /// Whether the timer has elapsed or not, true if yes.
+    fn poll_elapsed(&mut self, cx: &mut Context) -> Poll<bool>;
```

Comment:
What is the `bool` signifying here? Could it return `Poll<()>`?

Reply:
Whether the timer has elapsed yet. If we returned `()` we would not be able to distinguish between an instance where the timer was ready to say it hadn't elapsed yet, yes? Maybe that's a distinction that doesn't need to be made.

Reply:
I thought a timer not ready would return `Pending`. Would that not be equivalent to `Ready(false)` in your model?

Reply:
Turns out nothing. Resolved in f3c1b5b30